File: DogShelter/DogShelter.Model/PreporuceniPas.cs
namespace DogShelter.Model;

public class PreporuceniPas
{
    public int PasId { get; set; }
    public string Naziv { get; set; } = null!;
    public string RasaNaziv { get; set; } = null!;
    public string VelicinaNaziv { get; set; } = null!;
    public string SlikaNaslovna { get; set; } = null!;
    public DateOnly? DatumRodjenja { get; set; }
    public double Skor { get; set; }
    public string Razlog { get; set; } = null!;
    public bool Personalizovano { get; set; }
}

File: dogshelter_shared/lib/preporuke/domain/preporuceni_pas.dart
class PreporuceniPas {
  PreporuceniPas({
    required this.pasId,
    required this.naziv,
    required this.rasaNaziv,
    required this.velicinaNaziv,
    required this.slikaNaslovna,
    this.datumRodjenja,
    required this.skor,
    required this.razlog,
    required this.personalizovano,
  });

  final int pasId;
  final String naziv;
  final String rasaNaziv;
  final String velicinaNaziv;
  final String slikaNaslovna;
  final DateTime? datumRodjenja;
  final double skor;
  final String razlog;
  final bool personalizovano;

  factory PreporuceniPas.fromJson(Map<String, dynamic> json) {
    return PreporuceniPas(
      pasId: json['pasId'] as int,
      naziv: json['naziv'] as String,
      rasaNaziv: json['rasaNaziv'] as String,
      velicinaNaziv: json['velicinaNaziv'] as String,
      slikaNaslovna: json['slikaNaslovna'] as String,
      datumRodjenja: json['datumRodjenja'] == null ? null : DateTime.parse(json['datumRodjenja'] as String),
      skor: (json['skor'] as num).toDouble(),
      razlog: json['razlog'] as String,
      personalizovano: json['personalizovano'] as bool,
    );
  }
}
